rs_workflows/common.py

```diff
@@ -0,0 +1,451 @@
+"""Docstring will be here."""
+import enum
+import logging
+import sys
+import time
+from datetime import datetime
+
+import numpy as np
+import requests
+from prefect import exceptions, flow, get_run_logger, task
+from prefect_dask.task_runners import DaskTaskRunner
+
+CADIP = "CADIP"
+ADGS = "ADGS"
+
+DOWNLOAD_FILE_TIMEOUT = 180  # in seconds
+SET_PREFECT_LOGGING_LEVEL = "DEBUG"
+ENDPOINT_TIMEOUT = 2  # in seconds
+
+
+class EDownloadStatus(str, enum.Enum):
+    """
+    Download status enumeration.
+    """
+
+    NOT_STARTED = "NOT_STARTED"
+    IN_PROGRESS = "IN_PROGRESS"
+    FAILED = "FAILED"
+    DONE = "DONE"
+
+
+def get_general_logger(logger_name):
+    """Get a general logger with the specified name.
+
+    Args:
+        logger_name (str): The name of the logger.
+
+    Returns:
+        logging.Logger: A logger instance.
+    """
+
+    logger = logging.getLogger(logger_name)
+    logger.setLevel(logging.DEBUG)
+    logger.handlers = []
+    logger.propagate = False
+    console_handler = logging.StreamHandler(sys.stdout)
+    console_handler.setLevel(logging.DEBUG)
+    console_handler.setFormatter(logging.Formatter("[%(asctime)-20s] [%(name)-10s] [%(levelname)-6s] %(message)s"))
+    logger.addHandler(console_handler)
+    return logger
+
+
+def check_status(endpoint, filename):
+    """Check the status of a file download from the specified rs-server endpoint.
+
+    This function sends a GET request to the rs-server endpoint with the filename as a query parameter
+    to retrieve the status of the file download. If the response is successful and contains a 'status'
+    key in the JSON response, the function returns the corresponding download status enum value. If the
+    response is not successful or does not contain the 'status' key, the function returns a FAILED status.
+
+    Args:
+        endpoint (str): The rs-server endpoint URL to query for the file status.
+        filename (str): The name of the file for which to check the status.
+
+    Returns:
+        EDownloadStatus: The download status enum value based on the response from the endpoint.
+
+    """
+    # TODO: check the status for a certain timeout if http returns NOK ?
+    try:
+        response = requests.get(
+            endpoint,
+            params={"name": filename},
+            timeout=ENDPOINT_TIMEOUT,
+        )
+
+        eval_response = response.json()
+        if (
+            response.ok
+            and "name" in eval_response.keys()
+            and filename == eval_response["name"]
+            and "status" in eval_response.keys()
+        ):
+            return EDownloadStatus(eval_response["status"])
+
+    except requests.exceptions.RequestException as e:
+        print(f"Request exception caught when calling for status endpoint: {e}")
```

Comment:
I agree with Ovidiu, use logger instead of print

Reply:
Removed print